File: unity/Assets/Hallway/Constructor/RPG/firstVisitCourse.js
#pragma strict

var Bootstrap : GameObject;

function OnTriggerEnter() {
	var scr1 : StatisticParser = Bootstrap.GetComponent.<StatisticParser>();
	var scr2 : RPGParser = Bootstrap.GetComponent.<RPGParser>();
	if (!scr1.STAT.visited) {
		scr2.Achievement("Первое посещение курса!\n+10 очков!", 10);
		scr1.STAT.visited = true;
		scr1.Save();
	}
}

File: unity/Assets/Hallway/Constructor/RPG/firstVisitCourse.cs
using UnityEngine;

public class firstVisitCourse : MonoBehaviour
{
    GameObject Bootstrap;
    void OnTriggerEnter()
    {
        StatisticParser stParser = Bootstrap.GetComponent<StatisticParser>();
        RPGParser rpgParser = Bootstrap.GetComponent<RPGParser>();
        if (!stParser.stat.visited)
        {
            rpgParser.Achievement("Первое посещение курса!\n+10 очков!", 10);
            stParser.stat.visited = true;
            stParser.Save();
        }
    }
}
